x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -193,6 +195,91 @@ export const getTupleDuplicateErrorsAndUniqueRules = (
   return [Array.from(errors.values()), Array.from(rulesAcc.values())];
 };
 
+/**
+ * Query for a saved object with a given origin id and replace the
+ * id in the provided action with the _id from the query result
+ * @param action
+ * @param esClient
+ * @returns
+ */
+const swapActionIds = async (action: Action, esClient: ElasticsearchClient): Promise<Action> => {
+  try {
+    // actions are hidden SO types so we need to query
+    // with the elasticsearch client.. unless there
+    // is a better way to do this..
+    const actionWithRequestedOriginId = await esClient.search({
+      index: '.kibana',
+      body: {
+        query: {
+          term: {
+            originId: {
+              // we are attempting to import a rule that contains an action.
+              // That action (connector) has the old, pre-8.0 _id.
+              // We need to swap the old, pre-8.0 _id for the new id (alias_target_id)
+              // and replace the id of the action in the rule with the new _id
+              value: action.id,
+            },
+          },
+        },
+      },
+    });
+    if (actionWithRequestedOriginId.body.hits.hits.length === 1) {
+      // id is of the form 'action:1234-5678...' so I need to split off the 'action' prefix
+      return { ...action, id: actionWithRequestedOriginId.body.hits.hits[0]._id.split(':')[1] };
+    }
+  } catch (exc) {
+    return action;
```

Comment:
Not sure if the preference is to return the original action or an error? I would hate to blow things up unnecessarily. I felt the safest route was to just return the original action but if there are other suggestions I'm open to changing this.

Reply:
We discussed on a call but just posting this for posterity.

Agreed. I think we can do something similar to https://github.com/elastic/kibana/blob/ed77239d67071d247cdc00a4a3337041548a8e43/x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/import_rules_route.ts#L134-L135

Where we return both the error and the data and then stuff any import action errors into https://github.com/elastic/kibana/blob/ed77239d67071d247cdc00a4a3337041548a8e43/x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/import_rules_route.ts#L157

I'll do this in a follow-up PR 👍